### ISSUE TITLE
When training a single-output music separation model using ZFTurbo's MSST framework, the SDR approaches zero and the Loss hardly changes.

### ISSUE THREAD

**Issue by mechany** [ZFTurbo/Music-Source-Separation-Training#209]
When training a single-output music separation model using ZFTurbo's Music Source Separation Training framework, the SDR approaches zero and the Loss hardly changes. Under the same configuration, training a four-track separation model achieves an SDR of 10.1 on the validation set.
In the single-output training configuration:
instruments:
  - vocals
  - bass
  - drums
  - other
target_instrument: 'vocals'

In the four-stems training configuration:
instruments:
  - vocals
  - bass
  - drums
  - other
target_instrument: null
All other configurations are identical. The four-track model learns normally, whereas for the single-output model, the loss decreases extremely slowly and the SDR remains near zero.
Details: model name is  'mdx23'，other model appears the same problem

**Comment by mechany** [comment]
> Try again after replacing in models/mdx23c_tfc_tdf_v3.py, line 235:
> 
> ```
>         if self.num_target_instruments > 1:
>             b, c, f, t = x.shape
>             x = x.reshape(b, self.num_target_instruments, -1, f, t)
> ```
> 
> by
> 
> ```
>         b, c, f, t = x.shape
>         x = x.reshape(b, self.num_target_instruments, -1, f, t)
> ```

Looks very good, everything is getting better. Thank you for your help.
